### ISSUE TITLE
Add context menu in project explorer

### ISSUE THREAD

**Issue by rweickelt** [denis-shienkov/vscode-qbs#21]
Right-clicking on a product or sub-project in the project explorer should give a context menu with:

- Clean
- Build
- Run

for the respective product. That would allow the user to selectively build/rebuild a product.

**Comment by denis-shienkov** [comment]
I'm not sure about the `Run`. Because:

1. The current product can be not-runnable.
2. What to run if a user selected the 'sub-project'? (there are can be multiple runnable products)

**Comment by rweickelt** [comment]
1. Doesn't matter. In that case Qbs will simply fail I guess. But the question is whether this is the correct way to do it. Vscode rather uses launch.json files. 

2. Right, I'm case if sub projects there would be no run command. Only for products.

**Comment by denis-shienkov** [comment]
> Doesn't matter. In that case Qbs will simply fail I guess. But the question is whether this is the correct way to do it. Vscode rather uses launch.json files.

If we want to see the application output in the terminal, then we can't use the QBS to run the product. In this case creates the VSCode terminal using the [createTerminal()](https://code.visualstudio.com/api/references/vscode-api) VSCode API where are passes the product executable path && run env., ant the VSCode terminal homself starts the application.

**Comment by denis-shienkov** [comment]
@rweickelt , main question is: do we need to use the VSCode terminal for the product `Run` action, or not?

**Comment by rweickelt** [comment]
It would be nice to use the built-in terminal, but if that is very difficult, then maybe skip it for now.

> In this case creates the VSCode terminal using the createTerminal() VSCode API where are passes the product executable path && run env., ant the VSCode terminal homself starts the application.

I don't understand. I can only see that the createTerminal() API sets up a new terminal, but not that it starts any application. Where do you pass the executable?

**Comment by denis-shienkov** [comment]
Here is the snippet:

```
        const escaped = QbsUtils.escapeShell(executable);
        const terminal = vscode.window.createTerminal({
            name: 'QBS Run',
            env: env,
            cwd: path.dirname(executable)
        });
        if (process.platform === 'darwin') {
            // workaround for macOS system integrity protection
            const specialEnvs: string[] = ['DYLD_LIBRARY_PATH', 'DYLD_FRAMEWORK_PATH'];
            for (const specialEnv of specialEnvs) {
                if (env[specialEnv]) {
                    terminal.sendText('export ' + specialEnv + '=' + QbsUtils.escapeShell(env[specialEnv]));
                }
            }
        }
        terminal.sendText(escaped);
        terminal.show();
```

The `terminal.sendText(escaped)` is what do you need )).. And yes, it is crasy, but it is so (I'm too was confused..). ))

**Comment by rweickelt** [comment]
OK. And why is it not possible to send "qbs run -f PRODUCT -d BUILD_DIRECTORY" instead to the terminal? That way you could use the same terminal for all products (except on darwin where you need to set these variables, but that should be fixed in the Darwin cpp module), shouln't it.

**Comment by denis-shienkov** [comment]
> send "qbs run -f PRODUCT -d BUILD_DIRECTORY" instead to the terminal?

WOW,  I have not thought about it .. need  to check it, thanks.

**Comment by denis-shienkov** [comment]
Fixed in v2.0.0
